src/Kiota.Builder/Writers/Go/CodeMethodWriter.cs

```diff
@@ -442,29 +480,25 @@ private void WriteRequestBuilderBody(CodeClass parentClass, CodeMethod codeEleme
                     WriteReturnError(writer);
                     writer.DecreaseIndent();
                     writer.WriteLine("}");
+                    writer.DecreaseIndent(); //closing the scope for the err var
+                    writer.WriteLine("}");
                     return;
                 }
             }
-            switch(propertyType) {
-                case "string":
-                case "bool":
-                case "int32":
-                case "float32":
-                case "int64":
-                case "UUID":
-                case "Time": 
-                    writer.WriteLine($"{errorPrefix}WritePrimitiveValue({serializationKey}, {valueGet})");
-                    WriteReturnError(writer);
-                break;
-                default:
-                    writer.WriteLine($"{errorPrefix}WriteObjectValue({serializationKey}, {valueGet})");
-                    WriteReturnError(writer);
-                break;
+            var propertyType = conventions.TranslateType(propType, false);
+            if(conventions.IsPrimitiveType(propertyType) || conventions.IsScalarType(propertyType)) {
+                writer.WriteLine($"{errorPrefix}WritePrimitiveValue({serializationKey}, {valueGet})");
+                WriteReturnError(writer);
+            } else {
+                writer.WriteLine($"{errorPrefix}WriteObjectValue({serializationKey}, {valueGet})");
+                WriteReturnError(writer);
```

Comment:
You could move `WriteReturnError` outside and after the if/else and you won't have to repeat it in both blocks

Reply:
thanks, fixed.